R/get_smr.R

```diff
@@ -1,96 +1,162 @@
+#' get_smr(count = count, popn = popn, std_r = std_r, output_type = "counts")
+#' get_smr(count = count, popn = popn, std_r = std_r, percent = TRUE)
+#' get_smr(
+#'   count = count, popn = popn, std_r = std_r,
+#'   dist = "normal", interval = 0.95
+#' )
+#' get_smr(
+#'   count = count, popn = popn, std_r = std_r,
+#'   percent = TRUE, dist = "normal", interval = 0.95
+#' )
 #' }
 #'
 #' @importFrom magrittr %>%
+#' @importFrom rlang .data
 #' @export
-get_smr <- function(count, popn, std_rt, output_type = "ratio", percent = FALSE){
+get_smr <- function(count, popn, std_r, measure = "rate",
+                    output_type = "ratio", percent = FALSE,
+                    dist = NULL, interval = NULL) {
 
-  # validate inputs
+  # check validity of inputs
 
   if (!is.numeric(count)) {
-    stop("`count` must be numeric")
+    stop("`count` must be a non-negative integer of length 1")
   } else if (length(count) != 1) {
-    stop("`count` must be a scalar")
+    stop("`count` must be a non-negative integer of length 1")
+  } else if (count %% 1 != 0 | count < 0) {
+    stop("`count` must be a non-negative integer of length 1")
   }
 
-  if (length(popn) != length(std_rt)) {
-    stop("lengths of `popn` and `std_rt` must match")
+  if (length(popn) != length(std_r)) {
+    stop("lengths of `popn` and `std_r` must be the same")
   }
 
   if (!is.numeric(popn)) {
-    stop("`popn` must be a numeric vector")
-  } else if (!is.numeric(std_rt)) {
-    stop("`std_rt` must be a numeric vector")
+    stop("`popn` must be a numeric vector of non-negative integers")
+  } else if (any(popn < 0, na.rm = TRUE)) {
+    stop("`popn` must be a numeric vector of non-negative integers")
   }
 
-  if (!(output_type == "ratio" | output_type == "counts")) {
-    stop('`output_type` must be either "ratio" or "counts"')
+  # TODO: check if rates/risk are valid separately when risk SMR is added
+  if (!is.numeric(std_r)) {
+    stop("`std_r` must be a numeric vector")
   }
 
-  if (!percent %in% c(TRUE, FALSE)) {
-    stop("`percent` must be one of TRUE or FALSE")
+  # TODO: relax check when risk SMR is added
+  if (measure != "rate") {
+    stop("function currently only supports rate-SMR")
   }
 
-  if (output_type == "ratio") {
-    if (anyNA(popn)) {
-      stop("when calculating the SMR, `popn` should not have any NA values")
-    } else if (anyNA(std_rt)) {
-      stop("when calculating the SMR, `std_rt` should not have any NA values")
-    }
-  } else if (output_type == "counts") {
-    if (anyNA(popn)) {
-      warning("one or more elements in `popn` are NA")
-    } else if (anyNA(std_rt)) {
-      warning("one or more elements in `std_rt` are NA")
-    }
+  if (!output_type %in% c("ratio", "counts")) {
+    stop("`output_type` must be either 'ratio' or 'counts'")
+  }
+
+  if (!is.logical(percent)) {
+    stop("`percent` must be a boolean")
   }
 
-  exp_counts <- popn * std_rt
+  # calculate expected number of events in each stratum
+  expected <- popn * std_r
```

Comment:
Please put this in a if-block to calculated `expected` in this way only when `std_measure` is "rate".

If `std_measure` is "risk", then we need a new parameter passed in, `std_counts` and `expected <- std_counts * std_r` (see the SAS manual). But since we are not implementing it now, you can just implement the check if `std_measure` is "rate" and leave a TODO for adding a block for "risk"

Reply:
In rate: expected counts = population-time in stratum j * rate in stratum j
In risk: expected counts = # of observations in stratum k * risk in stratum j

Since we're not explicitly differentiating population-time versus # of observations for now - I don't think a separate if-else is needed to separate how we calculate expected counts

Reply:
Sounds good, thanks for explaining!